docs/src/styles/docs/algolia.scss

```diff
@@ -0,0 +1,104 @@
+.docs-header {
+  .DocSearch-Button {
+    display: inline-flex;
+    justify-content: center;
+    align-items: center;
+    border-radius: var(--amplify-components-button-border-radius);
+    box-sizing: border-box;
+    font-weight: 500;
+    transition: all var(--amplify-components-button-transition-duration);
+    background-color: inherit;
+    border-width: var(--amplify-components-button-border-width);
+    border-style: var(--amplify-components-button-border-style);
+    border-color: var(--amplify-colors-font-secondary);
+    border-radius: var(--amplify-components-button-border-radius);
+    color: var(--amplify-colors-font-secondary);
+    font-size: var(--amplify-components-button-font-size);
+    line-height: var(--amplify-components-button-line-height);
+    padding-block-start: var(--amplify-components-button-padding-block-start);
+    padding-block-end: var(--amplify-components-button-padding-block-end);
+    padding-inline-start: var(--amplify-components-button-padding-inline-start);
+    padding-inline-end: var(--amplify-components-button-padding-inline-end);
+    user-select: none;
+
+    &:hover {
+      background: var(--amplify-components-button-hover-background-color);
+      border-color: var(--amplify-components-button-hover-border-color);
+      color: var(--amplify-components-button-hover-color);
+      box-shadow: none;
+    }
+
+    :focus {
+      background-color: var(--amplify-components-button-hover-background-color);
+      border-color: var(--amplify-components-button-focus-border-color);
+      color: var(--amplify-components-button-focus-color);
+      box-shadow: var(--amplify-components-button-focus-box-shadow);
+    }
+
+    .DocSearch-Search-Icon {
+      color: inherit;
+    }
+  }
+
+  .DocSearch-Button-Key {
+    background: inherit;
+    color: inherit;
+  }
```

Comment:
We'll need to add this somewhere to get the height to line up with the color mode toggle

```scss
.DocSearch-Button-Placeholder {
  font-size: var(--amplify-components-button-small-font-size);
  line-height: 1;
}
```